cmake/BLTPrivateMacros.cmake

```diff
@@ -272,6 +272,46 @@ macro(blt_inherit_target_info)
 
 endmacro(blt_inherit_target_info)
 
+##------------------------------------------------------------------------------
+## blt_expand_depends( DEPENDS_ON [dep1 ...]
+##                     RESULT [variable] )
+##------------------------------------------------------------------------------
+macro(blt_expand_depends)
+    set(options)
+    set(singleValueArgs RESULT)
+    set(multiValueArgs DEPENDS_ON)
+
+    # Parse the arguments
+    cmake_parse_arguments(arg "${options}" "${singleValueArgs}"
+                        "${multiValueArgs}" ${ARGN} )
+
+    # Expand dependency list
+    set(_deps_to_process ${arg_DEPENDS_ON})
+    set(_expanded_DEPENDS_ON ${arg_DEPENDS_ON})
+    while(_deps_to_process)
+        # Copy the current set of dependencies to process
+        set(_current_deps_to_process ${_deps_to_process})
+        # and add them to the full expanded list
+        list(APPEND _expanded_DEPENDS_ON ${_deps_to_process})
+        # Then clear it so we can check if new ones were added
+        set(_deps_to_process)
```

Comment:
How are you guarding against circular dependencies causing an infinite loop?

Reply:
Dependencies are not added to `_expanded_DEPENDS_ON` if they're already in there, so if there exists a cycle (`A` depends on `B` and `B` depends on `A`), then it would terminate after adding `A` and `B` to the expanded dependencies.  That is, the `while` loop terminates after an iteration in which nothing new was added to the expanded dependencies.